cookbooks/fb_systemd/recipes/resolved.rb

```diff
@@ -31,10 +31,11 @@
 end
 
 # nss-resolve enables DNS resolution via the systemd-resolved DNS/LLMNR caching
-# stub resolver. According to upstream this should replace the glibc "dns"
-# resolver and is required for systemd-resolved to work. This block attempts
-# to place the resolver between mymachines and myhostname as recommended by
+# stub resolver. According to upstream this should be used in favor of the glibc
+# "dns" resolver and is required for systemd-resolved to work. This block attempts
+# to place "resolver dns" between mymachines and myhostname as recommended by
 # upstream.
+# See: https://man7.org/linux/man-pages/man8/nss-resolve.8.html
```

Comment:
Calling @davide125 and @anitazha to aisle 3...

I think this is probably fine, but there may be a performance or other reason to avoid that. If so, we should make it configurable.

Reply:
I did some spelunking and historically `resolve` was supposed to replace `dns`, which is why this was forcibly removing it. It looks like that guidance has now changed so inserting it before `dns` should be fine. I'm not in love with this whole logic as it's pretty brittle, but reworking it and adding some tests is probably something for another time.